### PR TITLE
Add tree to a commit object

### PR DESCRIPTION
**Pull request by alpacaaa**
Hey man, love this project!

I added a couple of things to the Github schema.
1. `tree` linked to a `Commit` object
2. `last_commit` in the `TreeEntry` object
3. `limit` as an argument of `commits`

Let me know if it looks good :)

### CODE REVIEW

File: graphqlhub-schemas/src/apis/github.js [hunk 1]
```diff
@@ -33,10 +33,14 @@ export let getReposForUser = (username) => {
   });
 };
 
-export let getCommitsForRepo = (username, reponame) => {
+export let getCommitsForRepo = (username, reponame, options = {}) => {
   let repo = github.getRepo(username, reponame);
+  let params = {};
+  if (options.limit) params.perpage = 1;
```
Comment (clayallsopp) [inline graphqlhub-schemas/src/apis/github.js]: should this be `params.perpage = options.limit`?

File: graphqlhub-schemas/src/github.js [hunk 2]
```diff
@@ -100,8 +151,8 @@ let IssueLabelType = new GraphQLObjectType({
   }
 });
 
-let grabUsernameAndReponameFromURL = (url) => {
-  let array = url.split('/repos/')[1].split('/issues')[0].split('/');
+let grabUsernameAndReponameFromURL = (url, resource) => {
+  let array = url.split('/repos/')[1].split('/' + resource)[0].split('/');
```
Comment (clayallsopp) [inline graphqlhub-schemas/src/github.js]: thanks for fixing this - can we make it possible without having to pass in the additional argument?

I think something like this works:

```
  let array = url.split('https://api.github.com/repos/')[1].split('/');
  return {
    username : array[0],
    reponame : array[1],
  };
```

**Comment by clayallsopp** [comment]
This is great, the tree view is super helpful - left some comments, but very awesome!

**Comment by alpacaaa** [comment]
Both your comments are right on, I'll fix those issues shortly! ;)

**Comment by alpacaaa** [comment]
There you go man, I also added `repo.owner` while I was there.

**Comment by clayallsopp** [comment]
:ship: should be live in a few minutes, thanks again!

**Comment by alpacaaa** [comment]
Great, thanks! :)

**Pull request merged**